File: leetcode/C++/add-strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int m=max(num1.length(),num2.length())-1, carry=0, n, p_n1=num1.length()-1, p_n2=num2.length()-1;
        string sol(m+1,'0');
        for(; p_n1>=0&&p_n2>=0; --p_n1, --p_n2, --m){
            n=carry+num1[p_n1]-'0'+num2[p_n2]-'0';
            carry=n/10;
            sol[m]=(n%10)+'0';
        }
        for(; p_n1>=0; --p_n1, --m){
            n=carry+num1[p_n1]-'0';
            carry=n/10;
            sol[m]=(n%10)+'0'; 
        }
        for(; p_n2>=0; --p_n2, --m){
            n=carry+num2[p_n2]-'0';
            carry=n/10;
            sol[m]=(n%10)+'0'; 
        }
        if(carry) sol="1"+sol;
        return sol;
    }
};

File: leetcode/Python/add-strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        m, p_n1, p_n2, carry = max(len(num1),len(num2)), len(num1)-1, len(num2)-1, 0
        sol = ['1']*(m+1)
        while p_n1>=0 and p_n2>=0:
            n=int(num1[p_n1])+int(num2[p_n2])+carry
            carry=n//10
            sol[m]= str(n%10)
            m, p_n1, p_n2 = m-1, p_n1-1, p_n2-1
        while p_n1>=0:
            n=int(num1[p_n1])+carry
            carry=n//10
            sol[m]= str(n%10)
            m, p_n1 = m-1, p_n1-1
        while p_n2>=0:
            n=int(num2[p_n2])+carry
            carry=n//10
            sol[m]= str(n%10)
            m, p_n2 = m-1, p_n2-1
        if carry==0:
            return ''.join(sol[1:])
        return ''.join(sol)